scenes/enemies/guard/guard.gd

```diff
@@ -0,0 +1,395 @@
+@tool
+class_name Guard extends CharacterBody2D
+## Enemy type that patrols along a path and raises an alert if the player is detected.
+
+## Emitted when the player is detected.
+signal player_detected(player: Node2D)
+
+enum State {
+	## Going along the path.
+	PATROLING,
+	## Player is in sight, it takes some time until the player is detected.
+	DETECTING,
+	## Player was detected.
+	ALERTED,
+	## Player was in sight, going to the last point where the player was seen.
+	INVESTIGATING,
+	## Lost track of player, walking back to the patrol path.
+	RETURNING
+}
+
+@export_category("Patrol")
+@warning_ignore("unused_private_class_variable")
```

Comment:
Can you explain why this warning ignore is needed?

Reply:
Yes, that variable is not used in the rest of the code, its only use is being shown as a button in the inspector thanks to `@export_tool_button`